hugegraph-core/src/main/java/com/baidu/hugegraph/backend/serializer/BytesBuffer.java

```diff
@@ -348,6 +352,24 @@ public Id readId(boolean big) {
         }
     }
 
+    public BytesBuffer writeIndexId(Id id) {
+        byte[] bytes = id.asBytes();
+        int len = bytes.length;
+        E.checkArgument(len > 0, "Can't write empty id");
+        E.checkArgument(len <= ID_MAX_LEN,
+                        "Id max length is %s, but got %s {%s}",
+                        ID_MAX_LEN, len, id);
+        this.write(bytes);
+        return this;
+    }
+
+    public BinaryId readIndexId() {
+        int b = this.peekLast();
+        int len = b & 0x7f;
```

Comment:
define const var ID_LEN_MASK like ID_MAX_LEN
and rename ID_MAX_LEN to ID_LEN_MAX